### ISSUE TITLE
[opencv4[qt]] build failure with latest commit version 0ab1a9e

### ISSUE THREAD

**Issue by imaqqqq** [microsoft/vcpkg#11138]
**Host Environment**
- OS: win10 1909 Build18363.815
- Compiler: VS2019 16.5.4

**To Reproduce**
`./vcpkg install opencv4[qt] --recurse`

**Failure logs** 
Computing installation plan...
The following packages will be rebuilt:
  * qt5[activeqt,core,declarative,essentials,imageformats,multimedia,networkauth,quickcontrols2,svg,tools]:x86-windows
The following packages will be built and installed:
    opencv4[core,dnn,jpeg,opengl,png,qt,tiff,webp]:x86-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/3: qt5:x86-windows
Removing package qt5:x86-windows...
Removing package qt5:x86-windows... done
Elapsed time for package qt5:x86-windows: 10.5 ms
Starting package 2/3: qt5:x86-windows
Building package qt5[activeqt,core,declarative,essentials,imageformats,multimedia,networkauth,quickcontrols2,svg,tools]:x86-windows...
-- Performing post-build validation
-- Performing post-build validation done
Building package qt5[activeqt,core,declarative,essentials,imageformats,multimedia,networkauth,quickcontrols2,svg,tools]:x86-windows... done
Installing package qt5[activeqt,core,declarative,essentials,imageformats,multimedia,networkauth,quickcontrols2,svg,tools]:x86-windows...
Installing package qt5[activeqt,core,declarative,essentials,imageformats,multimedia,networkauth,quickcontrols2,svg,tools]:x86-windows... done
Elapsed time for package qt5:x86-windows: 12.33 s
Starting package 3/3: opencv4:x86-windows
Building package opencv4[core,dnn,jpeg,opengl,png,qt,tiff,webp]:x86-windows...
Error: Building package opencv4:x86-windows failed with: CASCADED_DUE_TO_MISSING_DEPENDENCIES
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: opencv4:x86-windows
  Vcpkg version: 2020.02.04-nohash-external

Additionally, attach any relevant sections from the log files above.

**Additional context**
I removed all the qt5 stuff, and rebuilt from scratch, the same error.

**Comment by cenit** [comment]
I was able to reproduce your issue. There might be a circular reference inside features for qt5 CONTROL file @Neumann-A; this should be reproducible easily with an empty port, just depending on "qt5" .
Changing opencv4 dependency from qt5 to qt5-base seems to fix the issue, but I am not sure it is enough to satisfy dependencies

**Comment by Neumann-A** [comment]
Found in cmake\OpenCVFindLibsGUI.cmake:
```
if(WITH_QT)
  if(NOT WITH_QT EQUAL 4)
    find_package(Qt5 COMPONENTS Core Gui Widgets Test Concurrent REQUIRED NO_MODULE)
    if(Qt5_FOUND)
      set(HAVE_QT5 ON)
      set(HAVE_QT  ON)
      find_package(Qt5 COMPONENTS OpenGL QUIET)
      if(Qt5OpenGL_FOUND)
        set(QT_QTOPENGL_FOUND ON)
      endif()
    endif()
  endif()

  if(NOT HAVE_QT)
    find_package(Qt4 REQUIRED QtCore QtGui QtTest)
    if(QT4_FOUND)
      set(HAVE_QT TRUE)
    endif()
  endif()
endif()
```
Depending on only qt5-base seems correct if you ask me

**Comment by cenit** [comment]
Yes I just completed a build and it seems ok.
I am going to fix it in the next opencv 4.3 update.

Do you have time in any case to investigate circular dependency problems when a port file depends on qt5, since you are well into the matter? It seems that vcpkg in that case cannot resolve anymore the chain of dependencies and just throws hands up

**Comment by Neumann-A** [comment]
Since depending on the metaport `qt5` is probably always wrong having it fail feels more like a feature than a bug :). I look into it if I got some free time but it currently seems like a low priority issue. 

>It seems that vcpkg in that case cannot resolve anymore the chain of dependencies and just throws hands up

That seems like a bug which needs to be resolved in vcpkg sources.

**Comment by venabled** [comment]
This also breaks on macOS as well.

**Comment by venabled** [comment]
And confirmed that changing the dependency for the `qt` feature to point to `qt5-base` works on macOS
